### PR TITLE
[test] Remove `ifdef USE_MPI` from the MCH OpenACC/CUDA/MPI/C++ test

### PR DESCRIPTION
**Pull request by lxavier**
This reduce the complexity, and forces to compile with MPI

If refremce does not need a version that can run without MPI, I suggest to integrate this change, otherwise just close the PR

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

**Comment by lxavier** [comment]
yes we are only interested to run on at least 2 MPI threads you can remove the non mpi test

**Comment by vkarak** [comment]
@lxavier Thanks for your input.
@ajocksch Could you update the tests as well in this PR, so that we can merge?

**Comment by ajocksch** [comment]
I suggest to merge pull request #513 first. Then we should work on this one.

**Comment by vkarak** [comment]
@ajocksch If we merge, it will cause the current ReFrame test to fail. So this PR should fix the test as well.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 1]
```diff
@@ -2,12 +2,11 @@
...
 
-@rfm.parameterized_test([True], [False])
+@rfm.simple_test
 class OpenaccCudaCpp(rfm.RegressionTest):
-    def __init__(self, withmpi):
+    def __init__(self):
         super().__init__()
-        name_suffix = 'WithMPI' if withmpi else 'WithoutMPI'
-        self.name = 'OpenaccCudaCPP' + name_suffix
+        self.name = 'OpenaccCudaCPP'
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: You don't need this anymore.

Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: done

**Comment by vkarak** [comment]
@jenkins-cscs retry all

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**